common/utils/rawUnitConversion.ts

```diff
@@ -0,0 +1,24 @@
+import { fromTokenBase, getDecimalFromEtherUnit, TokenValue, Wei, UnitKey } from 'libs/units';
+
+interface Args {
+  value?: TokenValue | Wei | null;
+  symbol?: string;
+}
+
+interface EthProps extends Args {
+  unit: UnitKey;
+}
+interface TokenProps extends Args {
+  decimal: number;
+}
+
+// export const rawUnitConversion = (args: Args & (EthProps | TokenProps)) => {
+export const rawUnitConversion = (args: Args & (EthProps | TokenProps)) => {
```

Comment:
This should probably be in `libs/units` instead, and instead of re-writing the logic that the `UnitDisplay` component uses internally, you could pull out the logic within the component and re-use it in the lib.